Dropdown/index.jsx

```diff
@@ -249,6 +253,19 @@ const onMouseLeave = (component) => () =>
   })
 
 export default compose(
+  uncontrolled({
+    prop: 'focus',
+    defaultProp: 'autoFocus',
+    handlerName: 'onFocus',
```

Comment:
I think your proposal is a clear improvement over what we have today. This generalization feels less _ad hoc_, more future proof and easier to understand.

I will change the higher order components to this approach.